src/components/codecSupportHelper.ts

```diff
@@ -91,119 +295,285 @@ export function getMaxBitrateSupport(): number {
+    // taller resolutions.
+    //
+    // In these checks, we hold one resolution constant and expand the other to
+    // test.
+    const otherResolutions = [
+        // Wider resolutions
+
+        // 32:9 is a super ultrawide resolution typically used by monitors.
+        new Resolution(Math.floor(maxRes.height * 3.555), maxRes.height),
+
+        // 2.40:1 is used by some cinema shot on 35mm film.
+        new Resolution(Math.floor(maxRes.height * 2.4), maxRes.height),
+
+        // "21:9" is the marketing term for multiple ultrawide resolutions.
+        // The real aspect ratio is somewhere between 2.37:1 and 2.38:1.
+        new Resolution(Math.floor(maxRes.height * 2.37037), maxRes.height),
+
+        // 1.90:1 is a common IMAX resolution.
+        new Resolution(Math.floor(maxRes.height * 1.9), maxRes.height),
+
+        // 1.85:1 is sometimes used in Hollywood cinema.
+        new Resolution(Math.floor(maxRes.height * 1.85), maxRes.height),
+
+        // Taller resolutions.
+
+        // 9:19.5 is a common resolution for a horizontal modern phone.
+        new Resolution(maxRes.width, Math.floor(maxRes.width / 9) * 19.5),
+
+        // 9:16 is the vertical version of 16:9.
+        new Resolution(maxRes.width, Math.floor(maxRes.width / 9) * 16),
+
+        // 1:1 resolution
+        new Resolution(maxRes.width, maxRes.width),
+
+        // 4:3 is an older but still common resolution found on old TVs.
+        new Resolution(maxRes.width, Math.floor((maxRes.width / 4) * 3)),
+
+        // 16:10 is a common resolution for computer displays.
+        new Resolution(maxRes.width, Math.floor((maxRes.width / 16) * 10))
+    ];
+
+    for (const newRes of otherResolutions) {
+        if (
+            castContext.canDisplayType(
+                mimeType,
+                codecString,
+                newRes.width,
+                newRes.height
+            )
+        ) {
+            // Return early, since it'll be the best we'll find.
+            return newRes;
+        }
     }
 
-    return h264Profiles;
+    return maxRes;
 }
 
 /**
- * Get the highest H.264 level supported by the active Cast device.
- * @param deviceId - Cast device id.
- * @returns The highest supported H.264 level.
+ * Gets the supported profiles for a given video codec.
+ * @param codec - The video codec in question.
+ * @returns An array of the supported profiles.
  */
-export function getH264LevelSupport(deviceId: DeviceIds): number {
-    switch (deviceId) {
-        case DeviceIds.NESTHUBANDMAX:
-        case DeviceIds.GEN1AND2:
-            return 41;
-        case DeviceIds.GEN3:
-        case DeviceIds.ULTRA:
-            return 42;
-        case DeviceIds.CCGTV:
-            return 51;
-    }
-
-    return 0;
+export function getVideoProfileSupport(codec: VideoCodec): string[] {
+    const possibleProfiles = ((): string[] => {
+        switch (codec) {
+            case VideoCodec.H264:
+                return [
```

Comment:
These constants and the ones in `getCodecString` are very tightly coupled (as is the logic of formatting the string coupled to the base VideoCodec). Would it be better to actually have a derived class for each VideoCode variant that knows how to format it's own codec string, and exposes the underlying profiles supported?

Reply:
I think I understand what you mean, but do you mind providing a mini example?

FYI, this coupling starts to weaken with an upcoming change I'm stacking on top of this that adds support for checking BT.2020 and SMPTE ST 2084 support.